components/admin_console/team_channel_settings/channel/list/channel_list.tsx

```diff
@@ -60,37 +73,38 @@ export default class ChannelList extends React.PureComponent {
                 </div>
             </div>
         );
-    }
+    };
 
-    searchBarChangeHandler = (e) => {
+    private searchBarChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
         this.setState({searchString: e.target.value});
-    }
+    };
 
-    handleChannelSearchKeyUp = async (e) => {
+    private handleChannelSearchKeyUp = async (e: React.KeyboardEvent) => {
         const {key} = e;
         const {searchString} = this.state;
-
         if (key === Constants.KeyCodes.ENTER[0]) {
             if (searchString.length > 1) {
                 const response = await this.props.actions.searchAllChannels(searchString, '', false, 0, PAGE_SIZE);
-                this.setState({searchMode: true, channels: response.data.channels, searchTotalCount: response.data.total_count, pageResetKey: Date.now()});
+                if ('data' in response) {
+                    this.setState({searchMode: true, channels: response.data.channels, searchTotalCount: response.data.total_count, pageResetKey: Date.now()});
+                }
             }
         }
-
         if (searchString.length === 0) {
             this.resetSearch();
         }
-    }
-
-    getDataBySearch = async (page, perPage) => {
+    };
+    private getDataBySearch = async (page: number, perPage: number) => {
```

Comment:
@reflog  How would you convert this async function to TS when it has to be assigned to ```absProps.actions.getData```   defined at line 23?

Reply:
@allenlai18 - I've committed a possible fix. Let me know if this helps

Reply:
> @allenlai18 - I've committed a possible fix. Let me know if this helps

@reflog   This is a good solution and it simplifies it.  However, in index.ts mapDispatchToProps function wants the function interface to be the exact same from redux. How should I fix this?

Reply:
@allenlai18  committed suggested fix

Reply:
ah yes. that is a way of doing it.